core/domain/question_fetchers.py

```diff
@@ -63,14 +79,23 @@ def get_questions_and_skill_descriptions_by_skill_ids(
 
     for skill_ids_list in grouped_skill_ids:
         skills = skill_models.SkillModel.get_multi(skill_ids_list)
-        grouped_skill_descriptions.append(
-            [skill.description if skill else None for skill in skills])
+        skill_descriptions: List[Optional[str]] = []
+        for skill in skills:
+            if skill:
+                # Ruling out the possibility of None for mypy type checking.
+                assert isinstance(skill.description, str)
+                skill_descriptions.append(skill.description)
+            else:
+                skill_descriptions.append(None)
+    grouped_skill_descriptions.append(skill_descriptions)
```

Comment:
The backend tests are failing because `skill_descriptions` is referenced before assignment.
Shouldn't this be inside the for loop?

Reply:
Done